csi/src/dev.rs

```diff
@@ -0,0 +1,175 @@
+                "unsupported device scheme: {}",
+                scheme
+            ))),
+        }
+    }
+
+    /// Lookup an existing device in udev matching the given UUID
+    /// to obtain a device implementing the Detach trait.
+    pub async fn lookup(
+        uuid: &Uuid,
+    ) -> Result<Option<Box<dyn Detach>>, DeviceError> {
+        let iscsi_key: String = format!("nexus-{}", uuid.to_string());
+        let nvmf_key: String = format!("uuid.{}", uuid.to_string());
+
+        let mut enumerator = Enumerator::new()?;
+
+        enumerator.match_subsystem("block")?;
+        enumerator.match_property("DEVTYPE", "disk")?;
+
+        for device in enumerator.scan_devices()? {
+            if let Some(bus) = device.property_value("ID_BUS") {
+                if bus.to_str() == Some("scsi") {
+                    if let Some(model) = device.property_value("ID_MODEL") {
+                        if model.to_str() == Some("Nexus_CAS_Driver") {
+                            if let Some(value) =
+                                device.property_value("ID_SCSI_SERIAL")
+                            {
+                                if let Some(serial) = value.to_str() {
+                                    if iscsi_key.starts_with(serial) {
+                                        if let Some(value) =
+                                            device.property_value("ID_PATH")
+                                        {
+                                            if let Some(path) = value.to_str() {
+                                                if let Some(devname) = device
+                                                    .property_value("DEVNAME")
+                                                {
+                                                    if let Some(name) =
+                                                        devname.to_str()
+                                                    {
+                                                        let value = iscsi::IscsiDetach::from_path(name.to_string(), path)?;
+                                                        if value.uuid() == uuid
+                                                        {
+                                                            return Ok(Some(
+                                                                Box::new(value),
+                                                            ));
+                                                        }
+                                                    }
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    continue;
+                }
+            }
+
+            if let Some(wwn) = device.property_value("ID_WWN") {
```

Comment:
the same match_property can be used here and then we can jump straight to the grabbing of the ID_WWN